Caboodle/Shared/Exceptions.shared.cs

```diff
@@ -9,4 +9,12 @@ public NotImplentedInReferenceAssembly()
         {
         }
     }
+
+    public class PermissionException : Exception
```

Comment:
I think this should subclass UnauthorizedAccessException

Reply:
Love it!